designs/replicated-pv/mayastor/volume-clone.md

```diff
@@ -0,0 +1,198 @@
+---
+oep-number: OEP 3916
+title: Volume Cloning Feature for OpenEBS Mayastor
+authors:
+  - "@adam-lithus"
+owners:
+  - "@tiagolobocastro"
+  - "@adam-lithus"
+editor: TBD
+creation-date: 2025-04-30
+last-updated: 2025-04-30
+status: provisional
+---
+
+# Volume Cloning Feature for OpenEBS Mayastor
+
+## Table of Contents
+
+1. [Overview](#overview)
+2. [Motivation](#motivation)
+3. [Goals](#goals)
+4. [Non-Goals](#non-goals)
+5. [Proposal](#proposal)
+6. [User Stories](#user-stories)
+7. [Implementation Details](#implementation-details)
+8. [Testing](#testing)
+
+---
+
+## Overview
+
+This proposal introduces a volume cloning feature for Mayastor. A clone is a copy of
+an existing volume that can be used independently from the source volume. The feature
+will leverage Mayastor's existing snapshot and restore capabilities to efficiently
+implement cloning with copy-on-write (COW) functionality.
+
+Volume cloning will enable users to quickly create duplicates of volumes for various
+use cases such as development, testing, data analysis, and backup, without having to
+copy all the data.
+
+## Motivation
```

Comment:
IMO all of this is very noisy and just buries the lede. I'm not sure any of these motivations are particular to cloning, all of these use cases are solved with snapshots AFAICT. The whole point of #1718 is very particular to the Kubernetes CSI, where instead of going from PVC -> snapshot -> PVC we go directly from PVC -> PVC.

Reply:
Thanks for pointing this out @michaelbeaumont. I've updated this section to be more to the point.